modvendor/modvendor_test.go

```diff
@@ -15,18 +15,823 @@
+				"git::file://{{.}}/cool-module?ref=main#main.tf": Doc(
+					Module(
+						Labels("test"),
+						Str("source", "{{index . 4}}"),
+					),
+				),
+			},
+		},
+		{
+			name: "cycle: my-module -> (awesome-module -> cool-module -> my-module, best-module)",
+			layout: []string{
+				"g:my-module",
+				"g:awesome-module",
+				"g:best-module",
+				"g:cool-module",
+			},
+			source: "git::file://{{.}}/my-module?ref=main",
+			configs: []hclconfig{
+				{
+					repo: "my-module",
+					path: "my-module/main.tf",
+					data: Doc(
+						Module(
+							Labels("test"),
+							Str("source", "git::file://{{.}}/awesome-module?ref=main"),
+						),
+					),
+				},
+				{
+					repo: "my-module",
+					path: "my-module/other.tf",
+					data: Doc(
+						Module(
+							Labels("test"),
+							Str("source", "git::file://{{.}}/best-module?ref=main"),
+							Str("other", "value"),
+						),
+						Module(
+							Labels("test2"),
+							Str("source", "git::file://{{.}}/awesome-module?ref=main"),
+						),
+					),
+				},
+				{
+					repo: "awesome-module",
+					path: "awesome-module/main.tf",
+					data: Doc(
+						Module(
+							Labels("test"),
+							Str("source", "git::file://{{.}}/cool-module?ref=main"),
+						),
+					),
+				},
+				{
+					repo: "cool-module",
+					path: "cool-module/main.tf",
+					data: Doc(
+						Module(
+							Labels("test"),
+							Str("source", "git::file://{{.}}/my-module?ref=main"),
+						),
+					),
+				},
+			},
+			wantVendored: []string{
+				"git::file://{{.}}/my-module?ref=main",
+				"git::file://{{.}}/awesome-module?ref=main",
+				"git::file://{{.}}/cool-module?ref=main",
+				"git::file://{{.}}/best-module?ref=main",
+			},
+			wantFiles: map[vendorPathSpec]fmt.Stringer{
+				"git::file://{{.}}/my-module?ref=main#main.tf": Module(
+					Labels("test"),
+					Str("source", "{{index . 1}}"),
+				),
+				"git::file://{{.}}/my-module?ref=main#other.tf": Doc(
+					Module(
+						Labels("test"),
+						Str("source", "{{index . 3}}"),
+						Str("other", "value"),
+					),
+					Module(
+						Labels("test2"),
+						Str("source", "{{index . 1}}"),
+					),
+				),
+				"git::file://{{.}}/awesome-module?ref=main#main.tf": Doc(
+					Module(
+						Labels("test"),
+						Str("source", "{{index . 2}}"),
+					),
+				),
+			},
+		},
+	}
+
+	for _, tc := range tcases {
+		t.Run(tc.name, func(t *testing.T) {
+			s := sandbox.New(t)
+			s.BuildTree(tc.layout)
+
+			if tc.vendordir == "" {
+				tc.vendordir = "/vendor"
+			}
+
+			modulesDir := s.RootDir()
+			rootdir := t.TempDir()
+			for _, cfg := range tc.configs {
+				test.WriteFile(t, s.RootDir(), cfg.path,
+					fixupString(t, cfg.data.String(), modulesDir))
+
+				git := sandbox.NewGit(t, filepath.Join(modulesDir, cfg.repo))
+				git.CommitAll("files updated")
+			}
+			source := fixupString(t, tc.source, modulesDir)
+			modsrc, err := tf.ParseSource(source)
+			assert.NoError(t, err)
+			got := modvendor.Vendor(rootdir, tc.vendordir, modsrc)
```

Comment:
```suggestion
			for _, cfg := range tc.configs {
				test.WriteFile(t, s.RootDir(), cfg.path,
					fixupString(t, cfg.data.String(), modulesDir))

				git := sandbox.NewGit(t, filepath.Join(modulesDir, cfg.repo))
				git.CommitAll("files updated")
			}
			source := fixupString(t, tc.source, modulesDir)
			modsrc, err := tf.ParseSource(source)
			assert.NoError(t, err)
			
			rootdir := t.TempDir()
			got := modvendor.Vendor(rootdir, tc.vendordir, modsrc)
```

Move rootdir closer to where it is used, I was a little confused at first thinking it would be used on the for loop.